tasks/quality_gates.py

```diff
@@ -0,0 +1,144 @@
+    # Sort gates by error_types to group in between NoError, AssertionError and StackTrace
+    for gate in sorted(gate_states, key=lambda x: x["error_type"] is None):
+
+        def getMetric(metric_name, gate_name=gate['name']):
+            try:
+                return metric_handler.get_formated_metric(gate_name, metric_name)
+            except KeyError:
+                return "DataNotFound"
+
+        if gate["error_type"] is None:
+            body_info += f"|{SUCCESS_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            with_info = True
+        else:
+            body_error += f"|{FAIL_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            body_error_footer += f"|{gate['name']}|{gate['error_type']}|{gate['message'].replace("\n", "<br>")}|\n"
+            with_error = True
+
+    body_error_footer += "\n</details>\n"
+    body = f"Please find below the results from static quality gates\n{body_error+body_error_footer if with_error else ""}\n\n{body_info if with_info else ""}"
+
+    pr_commenter(ctx, title=title, body=body)
+
+
+def _print_quality_gates_report(gate_states: list[dict[str, typing.Any]]):
+    print(color_message("======== Static Quality Gates Report ========", "magenta"))
+    for gate in sorted(gate_states, key=lambda x: x["error_type"] is not None):
+        if gate["error_type"] is None:
+            print(color_message(f"Gate {gate['name']} succeeded {SUCCESS_CHAR}", "blue"))
+        elif gate["error_type"] == "AssertionError":
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} because of the following assertion failures :\n{gate['message']}",
+                    "orange",
+                )
+            )
+        else:
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} with the following stack trace :\n{gate['message']}",
+                    "orange",
+                )
+            )
+
+
+@task
+def parse_and_trigger_gates(ctx, config_path="test/static/static_quality_gates.yml"):
+    """
+    Parse and executes static quality gates
+    :param ctx: Invoke context
+    :param config_path: Static quality gates configuration file path
+    :return:
+    """
+    with open(config_path) as file:
+        config = yaml.safe_load(file)
+
+    gate_list = list(config.keys())
+    quality_gates_mod = __import__("tasks.static_quality_gates", fromlist=gate_list)
+    print(f"{config_path} correctly parsed !")
+    metric_handler = GateMetricHandler(
+        git_ref=os.environ["CI_COMMIT_REF_SLUG"], bucket_branch=os.environ["BUCKET_BRANCH"]
+    )
+
+    print(f"The following gates are going to run:\n\t- {"\n\t- ".join(gate_list)}")
+    final_state = True
```

Comment:
```suggestion
    is_success = True
```
so that on line 132 you would have `if not is_success` which is to me better than `if not final_state`
Besides I also suggest, when possible, to reverse the check, because it's always easier to read an affirmative sentence: `if is_success`